pkg/ccl/changefeedccl/event_processing.go

```diff
@@ -337,6 +360,26 @@ func (c *kvEventToRowConsumer) Close() error {
 	return nil
 }
 
+func (c *kvEventToRowConsumer) encodeForParquet(
+	ctx context.Context,
+	updatedRow cdcevent.Row,
+	topic TopicDescriptor,
+	updated, mvcc hlc.Timestamp,
+	alloc kvevent.Alloc,
+) error {
+	if sinkWithEncoder, ok := c.sink.(SinkWithEncoder); ok {
+		// Revisit: break the arguments into multiple lines
+		err := sinkWithEncoder.EncodeAndEmitRow(ctx, updatedRow, topic, updated, mvcc, alloc)
+		if err != nil {
+			return err
+		}
+	} else {
+		return errors.AssertionFailedf("Expected a sink with encoder for parquet format, found %T", c.sink)
```

Comment:
I would also change the structure of this if/else block to be more go-idiomatic:
```
sinkWithEncoder, ok := c.sink.(SinkWithEncoder)
if !ok {
   return ...
}
... Rest of the code
```
You might even find that you don't need to wrap that much :)